File: edukate-gateway/src/main/java/io/github/sanyavertolet/edukate/gateway/dtos/SignInRequest.java
package io.github.sanyavertolet.edukate.gateway.dtos;

import jakarta.validation.constraints.NotBlank;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;

@Data
@NoArgsConstructor
public class SignInRequest {
    @NotBlank
    private String username;

    @ToString.Exclude
    @NotBlank
    private String password;
}

File: edukate-gateway/src/main/kotlin/io/github/sanyavertolet/edukate/gateway/dtos/SignInRequest.kt
package io.github.sanyavertolet.edukate.gateway.dtos

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Size

data class SignInRequest(
    @field:NotBlank val username: String,
    @field:NotBlank @field:Size(max = 1000) val password: String,
) {
    override fun toString(): String = "SignInRequest(username=$username)"
}
